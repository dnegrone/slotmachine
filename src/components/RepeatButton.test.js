import React from 'react';
import { shallow } from 'enzyme';
import RepeatButton from './RepeatButton';

describe('Component Repeat Button <RepeatButton />', () => {
    const wrapper = shallow(<RepeatButton />);
    const mockCallBack = jest.fn();
    
    it('expects render a button', () => {
        // console.log(wrapper.debug());
        expect(wrapper.debug()).toMatchSnapshot();
    })

    it('expect button to have an id equals to repeatButton', () => {
        expect(wrapper.find('#repeatButton').length).toEqual(1);
    })

    it('expects render a button without text', () => {
        expect(wrapper.text()).toEqual('');
    })

    it('expect to have click event to undefined', () => {
        wrapper.find('button').simulate('click');
        expect(mockCallBack.mock.calls.length).toEqual(0);
    })
})