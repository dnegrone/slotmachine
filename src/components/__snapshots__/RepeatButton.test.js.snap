// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Repeat Button <RepeatButton /> expects render a button 1`] = `"<button aria-label=\\"Play again.\\" id=\\"repeatButton\\" onClick={[undefined]} />"`;
